src/ymfm.h

```diff
@@ -373,11 +373,11 @@ class ymfm_saved_state
 	void restore(uint32_t &data) { data = read(); data |= read() << 8; data |= read() << 16; data |= read() << 24; }
 	void restore(envelope_state &data) { data = envelope_state(read()); }
 	template<typename DataType, int Count>
-	void restore(DataType (&data)[Count]) { for (int index = 0; index < Count; index++) restore(data[index]); }
+	void restore(DataType (&data)[Count]) { for (size_t index = 0; index < Count; index++) restore(data[index]); }
 
 	// internal helper
 	ymfm_saved_state &write(uint8_t data) { m_buffer.push_back(data); return *this; }
-	uint8_t read() { return (m_offset < m_buffer.size()) ? m_buffer[m_offset++] : 0; }
+	uint8_t read() { return (m_offset < (int32_t) m_buffer.size()) ? m_buffer[m_offset++] : 0; }
 
```

Comment:
For numeric casts I prefer int32_t(val)-style casts instead of (int32_t)val.

Reply:
Any reason for C style casts instead of C++ style casts (i.e. `static_cast<int32_t>(val)`)?

Reply:
ok, using the former now

Reply:
@DeltaRazero no particular reason. I've moved away from C style casts in this PR and am now using `int32(val)`. Should I change that into a static_cast instead?

Reply:
`(int32_t)val` won't have any compile-time checks, but both `int32_t(val)` and `static_cast<int32_t>(val)` do, since they both result in direct initialization. So there's no difference between those two.
The only real difference is that I think that `static_cast` better conveys intention and is the more C++ way of dealing in this context, but it's subjective. I'll leave it up to Aaron to give his opinion about it.